src/test/java/org/springframework/data/couchbase/core/CouchbaseTemplateKeyValueIntegrationTests.java

```diff
@@ -94,8 +94,15 @@ void findByIdWithLock() {
 			);
 			assertTrue(exception.retryReasons().contains(RetryReason.KV_LOCKED), "should have been locked");
 		} finally {
-			sleepSecs(2);
-			couchbaseTemplate.removeByQuery(User.class).withConsistency(QueryScanConsistency.REQUEST_PLUS).all();
+			for(int i=0; i< 10; i++) {
+				sleepSecs(2);
+				try {
+					couchbaseTemplate.removeByQuery(User.class).withConsistency(QueryScanConsistency.REQUEST_PLUS).all();
+					break;
+				} catch (Exception e) {
+					e.printStackTrace(); // gives IndexFailureException if the lock is still active
```

Comment:
Does this hide other exceptions that aren't expected? (Sure, it logs the stack trace, but nobody's going to look at it if the test "passes".)